app/assets/javascripts/covid19/src/webapp/components/structures-table/CustomGridPagination.tsx

```diff
@@ -15,47 +16,50 @@ export interface CustomGridPaginationProps {
 
 export const CustomGridPagination: React.FC<CustomGridPaginationProps> = React.memo(props => {
     const { count, page, pageSize, pageSizes, setPage, setPageSize, isLoading } = props;
-    const [showInfo, { disable }] = useBooleanState(true);
+    const [showInfo, { disable: hideInfo }] = useBooleanState(true);
     const snackbar = useSnackbar();
+    const classes = useStyles();
 
-    const onPageChange = React.useCallback(
+    const maxPage = Math.ceil(count / (pageSize ?? 10));
+
+    const setPageFromEvent = React.useCallback(
         (_event: React.MouseEvent<HTMLButtonElement> | null, newPage: number) => {
             setPage(newPage);
         },
         [setPage]
     );
 
-    const onPageSizeChange = React.useCallback(
+    const changePageSize = React.useCallback(
         (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
             const pageSize = parseInt(event.target.value, 10);
             if (pageSize > 25 && showInfo) {
                 snackbar.info("Please note that larger page size may take longer to load.");
-                disable();
+                hideInfo();
             }
             setPageSize(pageSize);
         },
-        [setPageSize, snackbar, showInfo, disable]
+        [setPageSize, snackbar, showInfo, hideInfo]
     );
 
     return (
         <React.Fragment>
             <TablePagination
                 component="div" /* Default component is td, but we the parent component is not a table */
-                style={styles.table}
+                className={classes.table}
                 count={count}
                 page={page}
-                onPageChange={onPageChange}
+                onPageChange={setPageFromEvent}
                 rowsPerPageOptions={pageSizes}
                 rowsPerPage={pageSize || 10}
-                onRowsPerPageChange={onPageSizeChange}
-                backIconButtonProps={{ disabled: isLoading }}
-                nextIconButtonProps={{ disabled: isLoading }}
+                onRowsPerPageChange={changePageSize}
+                backIconButtonProps={{ disabled: isLoading || page === 0 }}
+                nextIconButtonProps={{ disabled: isLoading || page === maxPage - 1 }}
                 SelectProps={{ disabled: isLoading }}
```

Comment:
@tokland Oh okay, I see now. I'm sorry, I usually forget one of the reasons at a time, sometimes the immutability and other ones the performance. In the past, as you know I was used to putting `useMemo` everywhere, but later I did not use it that much for small conditions as it would not impact the performance, and I assumed that all objects and arrays would need `useMemo`. So here the point is to not give a new object each time to the component as it will re-render, so we want to keep props immutable. Thanks for the patience!!